packages/react-composites/src/composites/CallComposite/adapter/AzureCommunicationCallAdapter.ts

```diff
@@ -805,41 +826,38 @@ export const createAzureCommunicationCallAdapter = async ({
   return adapter;
 };
 
+/* @conditional-compile-remove(teams-identity-support) */
 /**
- * A custom React hook to simplify the creation of {@link CallAdapter}.
- *
- * Similar to {@link createAzureCommunicationCallAdapter}, but takes care of asynchronous
- * creation of the adapter internally.
- *
- * Allows arguments to be undefined so that you can respect the rule-of-hooks and pass in arguments
- * as they are created. The adapter is only created when all arguments are defined.
- *
- * Note that you must memoize the arguments to avoid recreating adapter on each render.
- * See storybook for typical usage examples.
- *
- * @public
+ * @beta
  */
-export const useAzureCommunicationCallAdapter = (
-  /**
-   * Arguments to be passed to {@link createAzureCommunicationCallAdapter}.
-   *
-   * Allows arguments to be undefined so that you can respect the rule-of-hooks and pass in arguments
-   * as they are created. The adapter is only created when all arguments are defined.
-   */
+export const createAzureCommunicationTeamsCallAdapter = async ({
+  userId,
+  credential,
+  locator,
+  /* @conditional-compile-remove(PSTN-calls) */ alternateCallerId
+}: AzureCommunicationCallAdapterArgs): Promise<TeamsCallAdapter> => {
+  const callClient = createStatefulCallClient({
+    userId,
+    /* @conditional-compile-remove(PSTN-calls) */ alternateCallerId
+  });
+  const callAgent = await callClient.createTeamsCallAgent(credential, {
+    undefined
+  });
+  const adapter = createAzureCommunicationTeamsCallAdapterFromClient(callClient, callAgent, locator);
+  return adapter;
+};
+
+/**
+ * @private
+ */
+const useAzureCommunicationCallAdapterGeneric = <
+  AdapterType extends CallAdapter | /* @conditional-compile-remove(teams-identity-support) */ TeamsCallAdapter
```

Comment:
Implement this without conditional compilation. It is internal only, and the type `TeamsCallAdapter` is available here in both build flavors.